### PR TITLE
[qa] Bug fixes and refactor

### PR DESCRIPTION
**Pull request by maflcko**
It is not possible to switch to py3 without fixing those bugs or doing the refactor. (E.g. python2 will silently cast floats to ints where needed, but I'd rather explicitly use ints)

To aid review, each commit in this pull will either fix bugs or do some refactor.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Extended test pass and enabling python deprecation warnings only give the following "false positive" warnings:

```
$ qa/pull-tester/rpc-tests.py -extended | grep -C 1 DeprecationWarning
/home/marco/workspace/bitcoin/qa/rpc-tests/test_framework/blocktools.py:49: DeprecationWarning: classic int division
  halvings = int(height/150) # regtest
/home/marco/workspace/bitcoin/qa/rpc-tests/replace-by-fee.py:33: DeprecationWarning: classic long division
  while node.getbalance() < satoshi_round((amount + fee)/COIN):
/home/marco/workspace/bitcoin/qa/rpc-tests/replace-by-fee.py:39: DeprecationWarning: classic long division
  txid = node.sendtoaddress(new_addr, satoshi_round((amount+fee)/COIN))
/home/marco/workspace/bitcoin/qa/rpc-tests/replace-by-fee.py:396: DeprecationWarning: classic long division
  split_value = int((initial_nValue-fee)/(MAX_REPLACEMENT_LIMIT+1))

File: qa/rpc-tests/test_framework/mininode.py [hunk 1]
```diff
@@ -192,13 +193,13 @@ def deser_string_vector(f):
...
+        r = struct.pack("B", len(l))
     elif len(l) < 0x10000:
-        r = chr(253) + struct.pack("<H", len(l))
+        r = struct.pack("B", 253) + struct.pack("<H", len(l))
     elif len(l) < 0x100000000L:
-        r = chr(254) + struct.pack("<I", len(l))
+        r = struct.pack("B", 254) + struct.pack("<I", len(l))
     else:
-        r = chr(255) + struct.pack("<Q", len(l))
+        r = struct.pack("B", 255) + struct.pack("<Q", len(l))
```
Comment (laanwj) [inline qa/rpc-tests/test_framework/mininode.py]: What about:
```
r = struct.pack("<BQ", 255, len(l))
```
etc?

Comment (MarcoFalke) [inline qa/rpc-tests/test_framework/mininode.py]: Sure, this makes sense. I've tried to make the diff easier to read but I will address you nit, as it makes the code smaller and possibly easier to read.

Comment (laanwj) [inline qa/rpc-tests/test_framework/mininode.py]: Thanks, yes I agree with regard to keeping the minimum diff, but indeed it's better to do this small cleanup (which someone is bound to do later) at the same time.

**Comment by laanwj** [comment]
Tested ACK 4444806

**Pull request merged**